.pre-commit-config.yaml

```diff
@@ -1,81 +1,110 @@
-        additional_dependencies:
+    - repo: https://github.com/PyCQA/autoflake
+      rev: v2.0.1
+      hooks:
+          - id: autoflake
+
+    - repo: https://github.com/asottile/pyupgrade
+      rev: v3.3.1
+      hooks:
+          - id: pyupgrade
+            args: [--py39-plus]
+
+    - repo: https://github.com/pycqa/isort
+      rev: 5.12.0
+      hooks:
+          - id: isort
+
+    - repo: https://github.com/RobertCraigie/pyright-python
+      rev: v1.1.298
+      hooks:
+          - id: pyright
+            entry: env PYRIGHT_PYTHON_FORCE_VERSION=latest pyright
+            args: [--pythonversion=3.11, --warnings]
+            additional_dependencies:
           # Required for pyright strict mode
-          - hypothesis
-          - hypothesmith
-          - pytest
-          - flake8
-          - trio
-          - anyio
+                - anyio
+                - flake8
+                - hypothesis
+                - hypothesmith
+                - pytest
+                - trio
 
--   repo: https://github.com/codespell-project/codespell
-    rev: v2.2.3
-    hooks:
-    - id: codespell
-      additional_dependencies:
-          - tomli
+    - repo: https://github.com/codespell-project/codespell
+      rev: v2.2.4
+      hooks:
+          - id: codespell
+            additional_dependencies:
+                - tomli
 
--   repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v4.4.0
-    hooks:
-    -   id: trailing-whitespace
-        args: ['--markdown-linebreak-ext=md,markdown']
-    -   id: end-of-file-fixer
-    -   id: fix-encoding-pragma
-        args: [--remove]
-    -   id: check-yaml
-    -   id: debug-statements
-        language_version: python3
+    - repo: https://github.com/pre-commit/pre-commit-hooks
+      rev: v4.4.0
+      hooks:
+          - id: check-merge-conflict
+          - id: check-toml
+          - id: check-yaml
+          - id: debug-statements
+            language_version: python3
+          - id: end-of-file-fixer
+          - id: fix-encoding-pragma
+            args: [--remove]
+          - id: trailing-whitespace
+            args: ['--markdown-linebreak-ext=md,markdown']
 
--   repo: https://github.com/PyCQA/flake8
-    rev: 6.0.0
-    hooks:
-    -   id: flake8
-        types: ["python", "pyi"]
-        language_version: python3
-        additional_dependencies:
-          - flake8-builtins
-          - flake8-bugbear
-          - flake8-comprehensions
-          - flake8-2020
-          - flake8-builtins
-          - flake8-bugbear
-          - flake8-comprehensions
-          - flake8-datetimez
-          - flake8-docstrings
-          - flake8-mutable
-          - flake8-noqa
-          - flake8-pie
-          - flake8-pytest-style
-          - flake8-return
-          - flake8-simplify
-          - flake8-type-checking
-          - flake8-pyi
+    - repo: https://github.com/PyCQA/flake8
+      rev: 6.0.0
+      hooks:
+          - id: flake8
+            types_or: [python, pyi]
```

Comment:
*very* important that this is `types_or` and not `types`, else it only checked stub files :sweat_smile: